src/components/enterprise-user-subsidy/tests/UserSubsidy.test.jsx

```diff
@@ -83,247 +50,67 @@ const UserSubsidyWithAppContext = ({
-        expect(fetchSubscriptionLicensesForUser).toHaveBeenCalledWith(TEST_ENTERPRISE_UUID);
-        expect(fetchCustomerAgreementData).toHaveBeenCalledWith(TEST_ENTERPRISE_UUID);
-        expect(requestAutoAppliedLicense).not.toHaveBeenCalled();
-      });
-    });
-
-    describe('with coupon codes, no license', () => {
-      beforeEach(() => {
-        fetchCouponCodeAssignments.mockResolvedValueOnce({
-          data: {
-            results: [{
-              code: 'test-code',
-              redemptions_remaining: 1,
-              catalog: 'test-catalog-uuid',
-            }],
-          },
-        });
-        fetchSubscriptionLicensesForUser.mockResolvedValueOnce(mockEmptyListResponse);
-        fetchCustomerAgreementData.mockResolvedValueOnce(mockEmptyListResponse);
-      });
-
-      afterEach(() => {
-        expect(fetchSubscriptionLicensesForUser).toHaveBeenCalledWith(TEST_ENTERPRISE_UUID);
-        expect(fetchCustomerAgreementData).toHaveBeenCalledWith(TEST_ENTERPRISE_UUID);
-        expect(fetchCouponCodeAssignments).toHaveBeenCalledWith({
-          enterprise_uuid: TEST_ENTERPRISE_UUID,
-          full_discount_only: 'True',
-          is_active: 'True',
-        });
-        expect(requestAutoAppliedLicense).not.toHaveBeenCalled();
-      });
-
-      test('has portal access and shows correct code redemptions', async () => {
-        const Component = (
-          <UserSubsidyWithAppContext>
-            <SubscriptionLicenseConsumer />
-            <CouponCodesConsumer />
-          </UserSubsidyWithAppContext>
-        );
-        renderWithRouter(Component, {
-          route: `/${TEST_ENTERPRISE_SLUG}`,
-        });
-
-        // assert component is initially loading
-        expect(screen.getByText(LOADING_SCREEN_READER_TEXT)).toBeInTheDocument();
-
-        await waitFor(() => {
-          expect(screen.queryByText('Coupon codes count: 1')).toBeInTheDocument();
-          expect(screen.queryByText('License status: none')).toBeInTheDocument();
-        });
-
-        // assert component is no longer loading
-        expect(screen.queryByText(LOADING_SCREEN_READER_TEXT)).not.toBeInTheDocument();
-      });
-    });
+  test.each([
+    {
+      isSubscriptionsLoading: false,
+      isCouponCodesLoading: false,
+      isEnterpriseOffersLoading: false,
+      isLoadingExpected: false,
+    },
+    {
+      isSubscriptionsLoading: true,
+      isCouponCodesLoading: false,
+      isEnterpriseOffersLoading: false,
+      isLoadingExpected: true,
+    },
+    {
+      isSubscriptionsLoading: false,
+      isCouponCodesLoading: true,
+      isEnterpriseOffersLoading: false,
+      isLoadingExpected: true,
+    },
+    {
+      isSubscriptionsLoading: false,
+      isCouponCodesLoading: false,
+      isEnterpriseOffersLoading: true,
+      isLoadingExpected: true,
+    },
+    {
+      isSubscriptionsLoading: true,
+      isCouponCodesLoading: true,
+      isEnterpriseOffersLoading: true,
+      isLoadingExpected: true,
+    },
```

Comment:
I wonder its worth doing one test case where 2 hooks are loading while one if false to verify it can still determine whether a its still resolving correctly.